src/dotnet-core-uninstall/Windows/RegistryQuery.cs

```diff
@@ -53,6 +35,30 @@ public virtual IEnumerable<Bundle> GetAllInstalledBundles()
             return wrappedBundles;
         }
 
+        private IEnumerable<RegistryKey> GetNetCoreBundleKeys(RegistryKey uninstallKey)
+        {
+            try
+            {
+                var uninstalls = uninstallKey
+                    .OpenSubKey("SOFTWARE")
+                    .OpenSubKey("Microsoft")
+                    .OpenSubKey("Windows")
+                    .OpenSubKey("CurrentVersion")
+                    .OpenSubKey("Uninstall");
+
+                var names = uninstalls.GetSubKeyNames();
+
+                return names
+                    .Select(name => uninstalls.OpenSubKey(name))
```

Comment:
Performance is probably not a concern, but each registry key you open will create an open file handle.

Reply:
Yeah it's not a big concern here. But is there a way to avoid opening these?

Reply:
For the first part, you could try something like OpenSubKey(Path.Combine("X,","Y", "Z"). For the second part, the only option there is to pull the data you need out of the registry keys and return it using a separate class that only has the properties you care about. It's probably not a big deal given that this runs on demand and only once each time you execute the tool

Reply:
These disposable objects end up leaking here if they fail the where clause.

Reply:
Makes sense, thanks!